Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -981,6 +1180,20 @@ void UpdateFeaturePointCloud() {
             throw std::runtime_error{"Tried to get non-existent plane."};
         }
 
+        Frame::Mesh& GetMeshByID(Frame::Mesh::Identifier meshID)
+        {
+            // Loop over the mesh vector and find the correct mesh.
+            for (Frame::Mesh& mesh : Meshes)
+            {
```

Comment:
We should consider using an unordered_map here instead of a vector. js/ts developers can directly call logic that will call this funciton. moving from n to log(n) would be great here.

I am not sure why plane detection uses a vector/do not have full context on that choice.

Reply:
The reason that we're using an std::Vector instead of a map is because IDs are created at initialization time for Planes/Meshs in xr.h, which is in line with how we handle native IDs for other XR objects like InputSource, SceneObject, etc.  So we don't have a way to get the ID before the object is created when calling emplace_back (or emplace if using a map).

We could switch to managing the IDs ourselves, but this would be a change in how we manage these IDs for other objects as well.  At the time we made the decision that the number of planes was small enough that it wasn't worth the tradeoff, but as we start introducing more and more of these it might make sense to make that change.

Eg.
`                struct Mesh
                {
                    using Identifier = size_t;
                    const Identifier ID{ NEXT_ID++ };
                private:
                    static inline Identifier NEXT_ID{ 0 };
                };
`